jerry-core/ecma/operations/ecma-regexp-object.c

```diff
@@ -1434,10 +1434,13 @@ ecma_regexp_exec_helper (ecma_value_t regexp_value, /**< RegExp object */
           capture_value = ecma_make_string_value (capture_str_p);
         }
 
+        ecma_property_descriptor_t array_item_prop_desc = ecma_make_empty_property_descriptor ();
+        array_item_prop_desc.is_value_defined = true;
+        array_item_prop_desc.value = capture_value;
         ECMA_TRY_CATCH (put_value,
-                        ecma_op_object_put (result_array_obj_p,
+                        ecma_op_object_define_own_property (result_array_obj_p,
                                             index_str_p,
-                                            capture_value,
+                                            &array_item_prop_desc,
                                             true),
```

Comment:
Oh, your new suggestion is more simpler than using define_own_property.
I will upload a new patch.

Just to be clear, as I understand, `ecma_free_value (capture_value)` can be removed because `ecma_set_named_data_property` *DOES* pass the ownership of `capture_value`, is it correct? Or are there any other reasons? I am little confused because in [jerryScript internals document](http://jerryscript.net/internals/), there's a statement like this: "When the value is passed to a function as an argument, the ownership of it will not pass, the called function have to make an own copy of the value".